graph/pkg/service/v0/ordering.go

```diff
@@ -45,3 +45,59 @@ func (s spacesByLastModifiedDateTime) Less(i, j int) bool {
 	// fallback to name if no dateTime is set on both items
 	return strings.ToLower(*s.spacesSlice[i].Name) > strings.ToLower(*s.spacesSlice[j].Name)
 }
+
+type userSlice []*libregraph.User
+
+// Len is the number of elements in the collection.
+func (d userSlice) Len() int { return len(d) }
+
+// Swap swaps the elements with indexes i and j.
+func (d userSlice) Swap(i, j int) { d[i], d[j] = d[j], d[i] }
+
+type usersByDisplayName struct {
+	userSlice
+}
+
+type usersByMail struct {
+	userSlice
+}
+
+type usersByOnPremisesSamAccountName struct {
+	userSlice
+}
+
+// Less reports whether the element with index i
+// must sort before the element with index j.
+func (u usersByDisplayName) Less(i, j int) bool {
+	return strings.ToLower(u.userSlice[i].GetDisplayName()) > strings.ToLower(u.userSlice[j].GetDisplayName())
```

Comment:
Are we sure the `>` is correct? Shouldn't it determine if `i < j`? (as the comment describes)

Reply:
:man_facepalming:

Reply:
@kobergj  PTAL  again. The actual behavior of the `$orderby` was correct. Just the semantics of the `Less()` methods were a little weird as you pointed out.